### PR TITLE
Changed ID generation logic to use AWS-provided ids

### PR DESCRIPTION
**Pull request by girodav**
## What does this PR do?

This PR changes the event ID generation logic for all inputs. The new logic combines different data provided by AWS (e.g S3 bucket-name, Kinesis Record Sequence Number, etc) to generate IDs. It removes any call to `hashlib.sha256` that is not really required and only adds further computational burden to the Lambda.
Integration tests were updated accordingly, to accommodate this change.

Docs will be updated as part of another PR: https://github.com/elastic/observability-docs/pull/2473 . However, a generic note about this change has been added to the README.

## Why is it important?

In its current implementation, the pool of possible IDs is too small to avoid duplicates. More details can be found at https://github.com/elastic/elastic-serverless-forwarder/issues/212

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.md`

## How to test this PR locally

`make unit-test`
`make integration-test`

## Related issues

- Closes https://github.com/elastic/elastic-serverless-forwarder/issues/212

### CODE REVIEW

File: handlers/aws/kinesis_trigger.py [hunk 1]
```diff
@@ -31,12 +31,14 @@ def _handle_kinesis_continuation(
...
 
     sequence_number = kinesis_record["kinesis"]["sequenceNumber"]
+    partition_key = kinesis_record["kinesis"]["partitionKey"]
     stream_type, stream_name, _ = get_kinesis_stream_name_type_and_region_from_arn(event_input_id)
 
     message_attributes = {
         "config": {"StringValue": config_yaml, "DataType": "String"},
         "originalStreamType": {"StringValue": stream_type, "DataType": "String"},
         "originalStreamName": {"StringValue": stream_name, "DataType": "String"},
+        "originalPartitionKey": {"StringValue": partition_key, "DataType": "String"},
```
Comment (aspacca) [inline handlers/aws/kinesis_trigger.py:41]: minor: let's add `partion_key` at https://github.com/elastic/elastic-serverless-forwarder/blob/462c5dbba65ed7a76975dd9c30dce962c1eb4c9d/handlers/aws/kinesis_trigger.py#L70 as well

**Review by aspacca — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -3,5 +3,9 @@
 # elastic-serverless-forwarder
 Elastic Serverless Forwarder
 
+## Important - v1.6.0
 
+#### Version 1.6.0 introduces a new way to generate event IDs. More information is available at [our troubleshooting documentation](https://www.elastic.co/guide/en/observability/master/aws-serverless-troubleshooting.html).
```
Comment (tommyers-elastic) [inline README.md:8]: this suggests to  me there there is some configuration available to generate new IDs. i would rephrase with something more definite e.g. "introduces a new event ID format which is backwards incompatible with previously indexed events; be aware that previously indexed events would be duplicated if they triggered the forwarder again after upgrading to this version".

**Review by tommyers-elastic — APPROVED** [review]
change looks good - one comment on the docs.

File: handlers/aws/kinesis_trigger.py [hunk 3]
```diff
@@ -122,6 +128,9 @@ def _handle_kinesis_record(
                             "name": stream_name,
                             "partition_key": kinesis_record["kinesis"]["partitionKey"],
                             "sequence_number": kinesis_record["kinesis"]["sequenceNumber"],
+                            "approximate_arrival_timestamp": int(
```
Comment (aspacca) [inline handlers/aws/kinesis_trigger.py:131]: let's put this in `meta.approximate_arrival_timestamp`, what do you think?

Comment (aspacca) [reply handlers/aws/kinesis_trigger.py:131]: I'd use milliseconds everywhere, since it's the greatest resolution we have across all inputs: 
```
int(float(kinesis_record["kinesis"]["approximateArrivalTimestamp"]) * 1000)
```

**Review by aspacca — COMMENTED** [review]
see comments

**Comment by girodav** [comment]
Last few changes, adding timestamps to the IDs, were made to comply with this issue https://github.com/elastic/integrations/issues/5100

File: README.md (hunk 2 above)
Comment (girodav) [reply README.md]: Good point 👍 

I already mentioned your last point in the troubleshooting documentation, but I agree that is best re-iterating it here, as it's the first thing users would see.

File: tests/handlers/aws/test_handler.py [hunk 4]
```diff
@@ -1764,57 +1769,73 @@ def test_lambda_handler_replay(self) -> None:
...
+        )
+
+        prefix_s3_first = f"{event_time}-{bucket_arn}-{first_filename}"
+        prefix_s3_second = f"{event_time}-{bucket_arn}-{second_filename}"
 
         _event_to_sqs_message(queue_attributes=self._queues_info["source-sqs-queue"], message_body=self._cloudwatch_log)
-        event_sqs = _event_from_sqs_message(queue_attributes=self._queues_info["source-sqs-queue"])
+        timestamp = str(int(datetime.datetime.utcnow().timestamp()))
+        queue_attributes = self._queues_info["source-sqs-queue"]
+        queue_attributes["sent_timestamp"] = timestamp
```
Comment (aspacca) [inline tests/handlers/aws/test_handler.py:1783]: I'm curious why this is needed at all: does not `sqs_client.receive_message()` include `messages["Messages"][]["attributes"]["SentTimestamp"]` in the first place?

Comment (aspacca) [reply tests/handlers/aws/test_handler.py:1783]: minor: replacing
[`messages = sqs_client.receive_message(QueueUrl=queue_attributes["QueueUrl"], MessageAttributeNames=["All"])`](https://github.com/elastic/elastic-serverless-forwarder/blob/a588ec0995d9a04aab916c81d64693568bb0b7e4/tests/handlers/aws/test_handler.py#L1235) with `messages = sqs_client.receive_message(QueueUrl=queue_attributes["QueueUrl"], AttributeNames=["All"], MessageAttributeNames=["All"])` will prevent the need for `queue_attributes["sent_timestamp"] = timestamp`

We can then have `_event_from_sqs_message` return `event_sqs` and `event_sent_timestamp`, similarly to what we do with `_event_from_cloudwatch_logs`

**Review by aspacca — APPROVED** [review]
minor refactoring of the test could be considered, not necessarily in the current PR

File: tests/handlers/aws/test_handler.py (hunk 4 above)
Comment (girodav) [reply tests/handlers/aws/test_handler.py]: You are totally right. I did not think about that :).

File: README.md [hunk 5]
```diff
@@ -3,5 +3,9 @@
 # elastic-serverless-forwarder
 Elastic Serverless Forwarder
 
+## Important - v1.6.0
```
Comment (aspacca) [inline README.md:6]: just realised this is not the readme published on SAR: rather the one at https://github.com/elastic/elastic-serverless-forwarder/blob/main/docs/README-AWS.md
